src/lib/seam/components/SeamEditableDeviceName/SeamEditableDeviceName.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable import/no-duplicates */
+
+import classNames from 'classnames'
+import type React from 'react'
+import { type KeyboardEvent, useCallback, useState } from 'react'
+
+import { CheckIcon } from 'lib/icons/Check.js'
+import { CloseIcon } from 'lib/icons/Close.js'
+import { EditIcon } from 'lib/icons/Edit.js'
+
+export type SeamDeviceNameProps = {
+  onEdit?: (newName: string) => void
+  editable?: boolean
+  tagName?: string
+  value: string
+} & React.HTMLAttributes<HTMLElement>
+
+function IconButton(
+  props: React.PropsWithChildren<React.HTMLAttributes<HTMLButtonElement>>
+): JSX.Element {
+  return (
+    <button
+      {...props}
+      className={classNames(
+        'seam-editable-device-name-icon-button',
+        props.className
+      )}
+    >
+      {props.children}
+    </button>
+  )
+}
+
+function fixName(name: string): string {
+  return name.replace(/\s+/g, ' ').trim()
+}
+
+function isValidName(
+  name: string
+): { type: 'success' } | { type: 'error'; message: string } {
```

Comment:
Kind of an arbitrary project convention, but we use `function` for components and arrow fns otherwise
```suggestion
const fixName = (name: string): string => {
  return name.replace(/\s+/g, ' ').trim()
}

const isValidName = (
  name: string
): { type: 'success' } | { type: 'error'; message: string } => {
```